lib/phlex/fifo.rb

```diff
@@ -1,34 +1,29 @@
 # frozen_string_literal: true
 
 class Phlex::FIFO
-	def initialize(max_bytesize)
+	def initialize(max_size)
 		@hash = {}
 		@mutex = Mutex.new
 
-		@bytesize = 0
-		@max_bytesize = max_bytesize
+		@size = 0
+		@max_size = max_size
 	end
 
-	attr_reader :bytesize, :max_bytesize
+	attr_reader :size, :max_size
 
 	def [](key)
-		@hash[key]
+		k, v = @hash[key.hash]
+		v if k == key
```

Comment:
Why not just use the Hash as the key and let Ruby do this? Because it’s quite a bit slower.